Document Cylc Review (minimally for next version)

sadielbartholomew opened the pull request:
First follow-on to #2614 (merge only after that PR is merged).

Start to add-in documentation on the Cylc Review service to the User Guide. Given we want to get this into the next version if possible, add only very basic information for no. Obviously this needs to be extended for the next version. ~~Additionally, add the associated CherryPy dependency to ``check-software``.~~[Edit: no need, CherryPy bundled with Cylc]

I've lifted & shifted Rose Bush documentation word-for-word, except for replacing with Cylc and Cylc Review etc., as appropriate, & referencing the screenshot as a figure, as is the style in the CUG. Let me know if you spot anything now inappropriate, though it seems fine as far as I can see.

Admittedly I have been struggling to find the necessary TeX dependencies to build the documentation on the Met Office software stacks, so couldn't build as a PDF. I'll try next week, but here is the essence for now.

hjoliver commented on doc/src/cylc-user-guide/cug.tex:
```diff
@@ -1203,6 +1203,80 @@ \subsection{Locating Suite And Task Output}
+\item Jinja2.
+
+\end {myitemize}
+
```
This probably should be just an comment on `cylc review`-specific requirements in the CUG installation section.

sadielbartholomew replied:
Yes, that is a much better location for it. Sorry for missing that.

sadielbartholomew commented:
Thanks Hilary, I've updated this in line with your comments (FYI, I didn't write it, I just lifted, shifted & search-&-replaced the text someone else had written about Rose Bush).

I also removed the extension to ``check-software`` as I recalled CherryPy is bundled with Cylc itself.
